delta-app/src/main/java/io/cdap/transformation/DefaultMutableRowSchema.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.transformation;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.transformation.api.MutableRowSchema;
+import io.cdap.transformation.api.NotFoundException;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of {@RowSchema}
+ */
+public class DefaultMutableRowSchema implements MutableRowSchema {
+
+  private Schema schema;
+  private Map<String, Schema.Field> fieldsMap;
+  private Map<String, String> originalToNewNames;
+  private Map<String, String> newToOriginalNames;
+  private boolean changed;
+
+  public DefaultMutableRowSchema(Schema schema) {
+    if (schema == null) {
+      throw new NullPointerException("Schema is null.");
```

Comment:
Better to throw IllegalArtumentException when input parameter is not per expectations, so caller can just handle it.